src/client/Client.cc

```diff
@@ -4345,6 +4385,7 @@ void Client::trim_caps(MetaSession *s, uint64_t max)
       if (all && in->ino != MDS_INO_ROOT) {
         ldout(cct, 20) << __func__ << " counting as trimmed: " << *in << dendl;
 	trimmed++;
+	_schedule_ino_release_callback(in.get());
```

Comment:
This is fine for release file inode, but it is not valid for releasing the directory inode.  When ganesha hold a ref (ll_ref =1) of directory inode, the dentry state is pinned and all=false,  although it can be released for ganesha, but here can not callbak. I have merged changes both of libcephfs and ganesha, the test found that the directory inode could not be released.
Does it fit here?